core/state/intra_block_state.go

```diff
@@ -450,9 +479,10 @@ func (sdb *IntraBlockState) Selfdestruct6780(addr libcommon.Address) {
 	if stateObject == nil {
 		return
 	}
-
 	if stateObject.newlyCreated {
-		sdb.Selfdestruct(addr)
+		if _, ok := types.ParseDelegation(sdb.GetCode(addr)); !ok {
```

Comment:
This interplay between EIP-7702 & EIP-6780 looks like a corner case that can be a source of tricky consensus bugs. Is it currently covered by the execution spec tests? We should also get it well specified in EIP-7702.

Reply:
it is covered in the tests (that's when I fixed it here).
I'll get a section added to the eip.

Reply:
https://github.com/ethereum/EIPs/pull/8832 cc: @somnathb1